src/models/beast.ts

```diff
@@ -16,20 +24,136 @@ export default class BeastModel {
 
             return beast;
         } catch (error: unknown) {
-            if (error instanceof Prisma.PrismaClientKnownRequestError) {
-                if (isErrorWithTarget(error)) {
-                    handleDatabaseError(error);
-                } else {
-                    throw new UnknownError(error.message);
-                }
-            } else if (error instanceof Prisma.PrismaClientValidationError) {
-                throw new UnprocessableEntityError(error.message);
+            BeastModel.handleErrors(error);
+        }
+        // this won't ever get hit, but eslint doesn't like when its not here for some reason lol
+        throw new UnknownError('An unexpected error occurred');
+    }
+
+    /**
+     * Get a beast by its id, also includes the beast's friends and preeshes made/received
+     * @param id Id of the requested beast
+     */
+    public static async getBeastById(id: number): Promise<Beast | null> {
+        try {
+            const beast = await prisma.beast.findUnique({
+                where: {
+                    id,
+                },
+                include: {
+                    beastieBros: true,
+                    authoredPreesh: {
+                        include: {
+                            comments: true,
+                        },
+                    },
+                    receivedPreesh: {
+                        include: {
+                            comments: true,
+                        },
+                    },
+                },
+            });
+            if (!beast) {
+                throw new NotFoundError(`Beast with id: ${id} does not exist.`);
+            }
+
+            return beast;
+        } catch (error: unknown) {
+            BeastModel.handleErrors(error);
+        }
+        throw new UnknownError('An unexpected error occurred');
+    }
+
+    /**
+     * Get a beast's friends by its id
+     * @param id The id of the beast
+     */
+    public static async getBeastFriends(id: number): Promise<Beast[] | null> {
```

Comment:
If the `getBeastById ` route gets the beast object as a whole, which includes friends' id's, is this route necessary? I feel like this could a function in the Beast client where we get our current beast, get a list of the ID's of friends, and then reuse the `getBeastById` route.

Reply:
Yea fair point, unnecessary I don't think there will be a case where we need to be accessing just the beast followers a bunch so, can remove if you guys both agree

Reply:
Yeah I think we can remove it. We can always add it if we think we need it

Reply:
Done